File: WebApp/Models/PoliceStation/PoliceStation.cs
using System.Collections.Generic;
using WebApp.Models.Shared;

namespace WebApp.Models.PoliceStation;

public class PoliceStation
{
    public string Id { get; set; }
    public string Name { get; set; }
    public string Url { get; set; }
    public Location Location { get; set; }
    public List<string> Services { get; set; }


    public override string ToString()
    {
        return $"{Id} {Name} {Url} {Location} {Services.ToArray()}";
    }

}

File: WebApp/wwwroot/js/PoliceStation.js
class PoliceStation {
    constructor(id, stationName, url, locationName, locationGps, services = []) {
        this.id = id;
        this.stationName = stationName;
        this.url = url;
        this.locationName = locationName;
        this.locationGps = locationGps;
        this.services = services;
    }
}

const mapOfStations = new Map();

function addStationToMap(PoliceStation) {
    mapOfStations.set(PoliceStation.id, PoliceStation);
}

export async function getAllStations() {
    const url = `${location.origin}/PoliceStations/GetAllPoliceStations`;

    try {
        const response = await fetch(url);
        if (!response.ok) {
            throw new Error(`HTTP error! status: ${response.status}`);
        }
        mapOfStations.clear();
        const data = await response.json();
        data.forEach(item => {
            const station = new PoliceStation(
                item.id,
                item.name,
                item.url,
                item.location.name,
                `${item.location.gpsLocation.latitude},${item.location.gpsLocation.longitude}`,
                item.services
            );
            addStationToMap(station);
        });
    } catch (error) {
        console.error('Error fetching police stations', error);
    }
}